pkg/agent/datamodel/helper.go

```diff
@@ -93,3 +94,25 @@ func IndentString(original string, spaces int) string {
 	}
 	return out.String()
 }
+
+func processContainerImage(downloadURL string) (string, error) {
+	// example URL "downloadURL": "mcr.microsoft.com/oss/kubernetes/autoscaler/addon-resizer:*",
+	// getting the data between the last / and the last :
+	parts := strings.Split(downloadURL, "/")
+	if len(parts) == 0 || len(parts[len(parts)-1]) == 0 {
+		return "", fmt.Errorf("container image component URL is not in the expected format: %s", downloadURL)
+	}
+	lastPart := parts[len(parts)-1]
+	component := strings.TrimSuffix(lastPart, ":*")
+	return component, nil
+}
+
+func processDownloadFile(downloadURL string) (string, error) {
```

Comment:
since this is re returning the name of the binary, can we rename to something like `getComponentNameFromURL`, and `getContainerImageNameFromURL`?